pkg/registry/query.go

```diff
@@ -95,15 +150,21 @@ func (q Querier) loadAPIBundle(k apiBundleKey) (*api.Bundle, error) {
 	return &b, nil
 }
 
-func (q Querier) ListPackages(_ context.Context) ([]string, error) {
+func (q *Querier) ListPackages(ctx context.Context) ([]string, error) {
+	if err := q.Wait(ctx); err != nil {
+		return nil, err
+	}
 	var packages []string
 	for pkgName := range q.pkgs {
 		packages = append(packages, pkgName)
 	}
 	return packages, nil
 }
 
-func (q Querier) ListBundles(ctx context.Context) ([]*api.Bundle, error) {
+func (q *Querier) ListBundles(ctx context.Context) ([]*api.Bundle, error) {
+	if err := q.Wait(ctx); err != nil {
```

Comment:
At the moment, gRPC client on OLM side doesn't have a deadline setting so the rRPC calls can wait here for a long time. I wonder if we should set a deadline now so that if it goes for too long, it will fail out with DEADLINE_EXCEED error. Or perhaps "waiting until ready regardless how long it is" is a desire behavior? I would love this to be tested in a real cluster to see if things work well.